### PR TITLE
feat(redis): enable TLS and prefix parameters

### PR DESCRIPTION
**Pull request by roemhildtg**
Some redis databases are configured to use TLS - this is especially common if you use the AWS MemoryDB version of redis. This PR allows you to configure the redis client to work with TLS and redis in cluster mode. 

`prefix` is required for cluster
`tls` is required for tls connections

#### I have verified this PR works with

-   [ ] Online form submission
-   [ ] Offline form submission
-   [ ] Saving offline drafts
-   [ ] Loading offline drafts
-   [ ] Editing submissions
-   [ ] Form preview
-   [ ] None of the above

#### What else has been done to verify that this works as intended?

- Tested with MemoryDB

#### Why is this the best possible solution? Were any other approaches considered?

 - This allows configuration of TLS/prefix in the app, but does not break existing configs

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

 - no affect

#### Do we need any specific form for testing your changes? If so, please attach one.

 - no

### CODE REVIEW

File: app/lib/db.js
Comment (lognaturel) [inline app/lib/db.js:9]: Do the `undefined` defaults work with a redis setup that doesn't use these? I want to make sure that folks happily using the current config will not need to make changes. 

Why not use `null` and `false` as defaults?

Comment (roemhildtg) [reply app/lib/db.js:9]: Good catch, I'm not sure why I used undefined. I suppose the undefined is not necessary - if a user doesn't provide anything it would be undefined (as it is currently) so I can just remove the || undefined and it would work for folks who don't use this.

Comment (roemhildtg) [reply app/lib/db.js]: I'm testing this fix - will resolve when I've verified it

Comment (roemhildtg) [reply app/lib/db.js]: Okay - I've squashed and pushed everything. I did test two scenarios:

1. default with no tls/no prefix enabled on a localhost
2. remote aws tls enabled redis server cluster - with tls and prefix passed. 

Both scenarios worked out for me. Ready for another review whenever there's some time to :)

Comment (lognaturel) [reply app/lib/db.js]: Thanks so much for giving it another look and making sure it works in both scenarios! We're wrapping up a few things before preparing a release and will try to get a final review and merge in before then.

If you ever customize widgets, you may want to look at https://github.com/enketo/enketo-express/issues/495#issuecomment-1499621829

Comment (roemhildtg) [reply app/lib/db.js]: Sounds good! Yeah, no rush, its better to have a full review than release a bug :+1:

**Review by eyelidlessness — APPROVED** [review]
Thanks!

**Review by eyelidlessness — CHANGES_REQUESTED** [review]
Actually, I ran this locally with a prefix for the cache DB, and was surprised to see a couple of test failures where the prefix appears to have been applied inconsistently across different Redis calls. I'm not sure what the appropriate solution is, but I'm putting this on hold for now so we can investigate further.

**Comment by roemhildtg** [comment]
> Actually, I ran this locally with a prefix for the cache DB, and was surprised to see a couple of test failures where the prefix appears to have been applied inconsistently across different Redis calls. I'm not sure what the appropriate solution is, but I'm putting this on hold for now so we can investigate further.

Good catch! I didn't realize that. How did you view the test failures?

**Comment by lognaturel** [comment]
We're completing a [monorepo migration](https://github.com/enketo/enketo-express/issues/552) and will be archiving this repository shortly. If you'd still like this change to be considered, please open a new PR at https://github.com/enketo/enketo If you're still not seeing the test failures that @eyelidlessness identified, we can talk about it there. Thanks!

**Pull request closed without merging**
